fmriprep/interfaces/utils.py

```diff
@@ -133,43 +135,55 @@ def _run_interface(self, runtime):
         return runtime
 
 
-def _tpm2roi(in_file, epi_mask, epi_mask_erosion_mm=0, erosion_mm=0, pthres=0.95):
+def _tpm2roi(t1_tpm, t1_mask, bold_mask, mask_erosion_mm=0, erosion_mm=0, pthres=0.95):
     """
     Generate a mask from a tissue probability map
     """
-    probability_map_nii = resample_to_img(in_file, epi_mask)
-    probability_map_data = probability_map_nii.get_data()
-
-    # thresholding
-    probability_map_data[probability_map_data < pthres] = 0
-    probability_map_data[probability_map_data != 0] = 1
-
-    epi_mask_nii = nb.load(epi_mask)
-    epi_mask_data = epi_mask_nii.get_data()
-    if epi_mask_erosion_mm > 0:
-        epi_mask_data = nd.binary_erosion(
-            epi_mask_data,
-            iterations=int(epi_mask_erosion_mm /
-                           max(probability_map_nii.header.get_zooms()))).astype(int)
-        eroded_mask_file = os.path.abspath("erodd_mask.nii.gz")
-        nb.Nifti1Image(epi_mask_data, epi_mask_nii.affine,
-                       epi_mask_nii.header).to_filename(eroded_mask_file)
-    else:
-        eroded_mask_file = epi_mask
-    probability_map_data[epi_mask_data != 1] = 0
+    tpm_img = nb.load(t1_tpm)
+    bold_mask_img = nb.load(bold_mask)
+
+    probability_map_data = (tpm_img.get_data() >= pthres).astype(np.uint8)
+
+    eroded_mask_file = bold_mask
+    if mask_erosion_mm > 0:
+        eroded_mask_file = os.path.abspath("eroded_mask.nii.gz")
+
+        mask_img = nb.load(t1_mask)
+        iter_n = int(mask_erosion_mm / max(mask_img.header.get_zooms()))
```

Comment:
Time isn't really an issue, but I'll give the ball approach a shot. It seems reasonable.